compositor_render/src/gpu_shader.rs

```diff
@@ -114,7 +114,7 @@ impl GpuShader {
     pub fn render(
         &self,
         params: &wgpu::BindGroup,
-        sources: &[(&NodeId, &NodeTexture)],
+        textures: &[Option<&Texture>],
```

Comment:
This is a very large change, and I think it's done for the wrong reasons. You are changing it just because it does not fit the API you want for web renderer and this requires addinging workarounds to all other components. This is not a good tradeoff.

I see 2 approaches:
- Ideally, any changes you are doing here should not require any logic changes in other components. You can achieve that by implementing other render method and reusing some code.
- You could not reuse this Shader struct
- You could change this API to accept Option<&Texture>, but calling `sources_to_textures` in every component is not a good solution